engine/src/main/java/org/terasology/math/AABB.java

```diff
@@ -146,15 +146,56 @@ public AABB move(Vector3f offset) {
     }
 
     public AABB transform(Quat4f rotation, Vector3f offset, float scale) {
-        Transform transform = new Transform(new Matrix4f(VecMath.to(rotation), VecMath.to(offset), scale));
+        Transform transform = new Transform(offset, rotation, scale);
         return transform(transform);
     }
 
     public AABB transform(Transform transform) {
-        javax.vecmath.Vector3f newMin = new javax.vecmath.Vector3f();
-        javax.vecmath.Vector3f newMax = new javax.vecmath.Vector3f();
-        AabbUtil2.transformAabb(VecMath.to(min), VecMath.to(max), 0.01f, transform, newMin, newMax);
-        return new AABB(VecMath.from(newMin), VecMath.from(newMax));
+        return transform(transform, DEFAULT_MARGIN);
+    }
+
+    public AABB transform(Transform transform, float margin) {
+        // Adaptation of AabbUtil2.transformAabb in the TeraBullet
```

Comment:
`// Adaptation of method AabbUtil2.transformAabb from the TeraBullet library. `

(remember the people reading your code in the future might not yet know what kind of animal TeraBullet is)

Reply:
I understand. Would `from MovingBlocks' fork of JBullet` be better?

Reply:
Nah. In the context of Terasology project we talk about TeraBullet. In the context of the TeraBullet project we can talk about it being a fork.

Reply:
What do you recommend?

Reply:
// Adaptation of method AabbUtil2.transformAabb from the TeraBullet library.

as stated above. :wink:

Reply:
Lol. I thought you were simply restating what I wrote but didn't realize that you were suggesting a change 🙄